# Exploratory Data Analysis - Statistik Dasar
Minggu 6 - Hari Senin

Menganalisis karakteristik dataset Titanic menggunakan statistik deskriptif.

In [1]:
import pandas as pd

df = pd.read_csv('Final_Clean_Dataset_titanic.csv')
df.head()

,Passenger_Id,Survived,P_Class,Name,Sex,Age,Sib_Sp,Parch,Ticket,Fare,Cabin,Embarked,Family_Size,Is_Alone,Age_Group,Title
0,1,0,3,"Braund, Mr. Owen Harris",MALE,22.0,1,0,A/5 21171,7.25,Unknown,S,2,0,Dewasa,Mr
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",FEMALE,38.0,1,0,PC 17599,71.28,C85,C,2,0,Dewasa,Mrs
2,3,1,3,"Heikkinen, Miss. Laina",FEMALE,26.0,0,0,STON/O2. 3101282,7.92,Unknown,S,1,1,Dewasa,Miss
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",FEMALE,35.0,1,0,113803,53.10,C123,S,2,0,Dewasa,Mrs
4,5,0,3,"Allen, Mr. William Henry",MALE,35.0,0,0,373450,8.05,Unknown,S,1,1,Dewasa,Mr


## 1. describe() - Statistik Deskriptif Kolom Numerik

In [2]:
df[['Age', 'Fare', 'Family_Size', 'Sib_Sp', 'Parch']].describe()

,Age,Fare,Family_Size,Sib_Sp,Parch
count,891.000000,891.000000,891.000000,891.000000,891.000000
mean,29.361616,32.204366,1.904602,0.523008,0.381594
std,13.019622,49.693414,1.613459,1.102743,0.806057
min,0.400000,0.000000,1.000000,0.000000,0.000000
25%,22.000000,7.910000,1.000000,0.000000,0.000000
50%,28.000000,14.450000,1.000000,0.000000,0.000000
75%,35.000000,31.000000,2.000000,1.000000,0.000000
max,80.000000,512.330000,11.000000,8.000000,6.000000


## 2. Statistik Dasar Satu per Satu (Mean, Median, Modus, Min, Max, Std, Kuartil)

In [3]:
numeric_cols = ['Age', 'Fare', 'Family_Size', 'Sib_Sp', 'Parch']

for col in numeric_cols:
    s = df[col]
    print(f"--- {col} ---")
    print(f"Mean       : {s.mean():.2f}")
    print(f"Median     : {s.median():.2f}")
    print(f"Modus      : {s.mode().iloc[0]:.2f}")
    print(f"Minimum    : {s.min():.2f}")
    print(f"Maksimum   : {s.max():.2f}")
    print(f"Std Deviasi: {s.std():.2f}")
    print(f"Q1 (25%)   : {s.quantile(0.25):.2f}")
    print(f"Q3 (75%)   : {s.quantile(0.75):.2f}")
    print()

--- Age ---
Mean       : 29.36
Median     : 28.00
Modus      : 28.00
Minimum    : 0.40
Maksimum   : 80.00
Std Deviasi: 13.02
Q1 (25%)   : 22.00
Q3 (75%)   : 35.00

--- Fare ---
Mean       : 32.20
Median     : 14.45
Modus      : 8.05
Minimum    : 0.00
Maksimum   : 512.33
Std Deviasi: 49.69
Q1 (25%)   : 7.91
Q3 (75%)   : 31.00

--- Family_Size ---
Mean       : 1.90
Median     : 1.00
Modus      : 1.00
Minimum    : 1.00
Maksimum   : 11.00
Std Deviasi: 1.61
Q1 (25%)   : 1.00
Q3 (75%)   : 2.00

--- Sib_Sp ---
Mean       : 0.52
Median     : 0.00
Modus      : 0.00
Minimum    : 0.00
Maksimum   : 8.00
Std Deviasi: 1.10
Q1 (25%)   : 0.00
Q3 (75%)   : 1.00

--- Parch ---
Mean       : 0.38
Median     : 0.00
Modus      : 0.00
Minimum    : 0.00
Maksimum   : 6.00
Std Deviasi: 0.81
Q1 (25%)   : 0.00
Q3 (75%)   : 0.00



## 3. value_counts() - Distribusi Kolom Kategorikal

In [4]:
print(df['Sex'].value_counts())
print()
print(df['Embarked'].value_counts())
print()
print(df['P_Class'].value_counts())

Sex
MALE      577
FEMALE    314
Name: count, dtype: int64

Embarked
S    646
C    168
Q     77
Name: count, dtype: int64

P_Class
3    491
1    216
2    184
Name: count, dtype: int64


In [5]:
print(df['Age_Group'].value_counts())
print()
print(df['Title'].value_counts())
print()
print(df['Survived'].value_counts())

Age_Group
Dewasa    752
Anak       69
Remaja     44
Lansia     26
Name: count, dtype: int64

Title
Mr              517
Miss            182
Mrs             125
Master           40
Dr                7
Rev               6
Major             2
Mlle              2
Col               2
Don               1
Mme               1
Ms                1
Lady              1
Sir               1
Capt              1
the Countess      1
Jonkheer          1
Name: count, dtype: int64

Survived
0    549
1    342
Name: count, dtype: int64


## 4. unique() dan nunique() - Nilai Unik per Kolom

In [6]:
categorical_cols = ['Sex', 'Embarked', 'P_Class', 'Age_Group', 'Title', 'Survived']

for col in categorical_cols:
    print(f"{col}: {df[col].nunique()} nilai unik -> {df[col].unique().tolist()}")

Sex: 2 nilai unik -> ['MALE', 'FEMALE']
Embarked: 3 nilai unik -> ['S', 'C', 'Q']
P_Class: 3 nilai unik -> [3, 1, 2]
Age_Group: 4 nilai unik -> ['Dewasa', 'Anak', 'Remaja', 'Lansia']
Title: 17 nilai unik -> ['Mr', 'Mrs', 'Miss', 'Master', 'Don', 'Rev', 'Dr', 'Mme', 'Ms', 'Major', 'Lady', 'Sir', 'Mlle', 'Col', 'Capt', 'the Countess', 'Jonkheer']
Survived: 2 nilai unik -> [0, 1]


## 5. Ringkasan Awal

- Rata-rata umur penumpang sekitar 29 tahun, dengan sebaran (std) sekitar 13 tahun.
- Rata-rata harga tiket (Fare) 32.2, tapi mediannya cuma 14.45 — artinya distribusinya miring ke kanan (ada beberapa tiket yang jauh lebih mahal, misalnya di kelas 1).
- Mayoritas penumpang laki-laki (577) dibanding perempuan (314).
- Mayoritas penumpang berangkat dari Southampton (S) - 646 orang.
- Kelas 3 adalah kelas dengan penumpang terbanyak (491 orang).
- Title terbanyak adalah 'Mr' (517), diikuti 'Miss' (182) dan 'Mrs' (125).
- Dari 891 penumpang, hanya 342 (38.4%) yang selamat.